azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -153,19 +153,24 @@ protected void deployArtifacts(WebAppConfiguration webAppConfiguration)
                     Log.warn("No <resources> is found in <deployment> element in pom.xml, skip deployment.");
                     return;
                 }
-                handleV2Resources(target, v2Resources);
+                final List<Resource> nonExternalResources = handleV2Resources(target, v2Resources);
+                if (nonExternalResources == null || nonExternalResources.isEmpty()) {
+                    Log.info("All external resources are already deployed.");
+                    return;
+                }
             }
             artifactHandler.publish(target);
         } finally {
             util.afterDeployArtifacts();
         }
     }
 
-    private void handleV2Resources(final DeployTarget target, List<Resource> v2Resources) throws IOException, AzureExecutionException {
+    private List<Resource> handleV2Resources(final DeployTarget target, List<Resource> v2Resources) throws IOException, AzureExecutionException {
```

Comment:
We may need to rename this method as a handle method should not return a List with specified values, or we could move `artifactHandler.publish(target);` to handle methods